rpc/sync.go

```diff
@@ -0,0 +1,18 @@
+package rpc
+
+import "github.com/NethermindEth/juno/core/felt"
+
+type SyncState struct {
+	False  *bool       `json:"false,omitempty"`
```

Comment:
Any particular reason this a pointer?

Reply:
We want the field to be omitted from the JSON representation if its value is nil (i.e. when the value is not false). If False was defined as a regular bool field instead of a pointer to a bool, it would be included in the JSON representation with a value of `false` even if it was not set or explicitly set to `false`. 

That was what I discovered when I tried it with a regular `bool`. Since we don't want that to happen according to the [spec](https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L878), I had to change it.